reconcile/utils/terraform_resource_spec.py

```diff
@@ -0,0 +1,88 @@
+from dataclasses import dataclass, field
+import json
+from typing import Any, Optional, cast
+
+
+@dataclass
+class TerraformResourceSpec:
+
+    resource: dict[str, Any]
```

Comment:
I prefer property declaration as done in `TerraformResourceIdentifier`. Is there a reason for the diff in implementation?

Reply:
consider this the minimum viable product to represent a terraform resource spec from the [schema](https://github.com/app-sre/qontract-schemas/blob/8ebce0bf85270b09c0919e0195fb1fc2ba424f44/graphql-schemas/schema.yml#L968) . this enables us work with the rich context this dataclass provides without going all in for
* mapping out the sub dataclasses
* adapting terrascript client
this would have widened the scope of this PR, which is desired state and vault management.

this is something for upcoming refactorings for sure!